### PR TITLE
Update gutenberg to v1.10.3

### PR DESCRIPTION
**Pull request by etoledom**
WPiOS side of https://github.com/wordpress-mobile/WordPress-Android/pull/10389
Implements the new gutenberg hotfix version for parity with Android.
This hotfix does not have any change on the iOS side.

https://github.com/wordpress-mobile/gutenberg-mobile/pull/1293

To test:
- Build and run.
- Smoke test gutenberg.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by hypest** [comment]
Similar to the Android side, I'm adding @marecar3 as a reviewer to this too.

**Comment by marecar3** [comment]
Thanks, @hypest!
I will take it from here 👍

File: Podfile.lock [hunk 1]
```diff
@@ -47,7 +47,7 @@ PODS:
   - "GoogleToolboxForMac/NSString+URLArguments (2.2.1)"
   - Gridicons (0.18)
   - GTMSessionFetcher/Core (1.2.2)
-  - Gutenberg (1.10.2):
+  - Gutenberg (1.10.3):
```
Comment (marecar3) [inline Podfile.lock]: Not sure how why Podfile.lock has 1.10.3 before we released it?  @etoledom

Comment (etoledom) [reply Podfile.lock]: Cocoapods reads the pod version from the Podspec file on `gutenberg-mobile`. And the Podspec gets the version from `package.json`. 

So, as soon as we change the `package.json` version number, cocoapods will know what `gutenberg-mobile` version we are installing 🎉

**Comment by etoledom** [comment]
Thank you @marecar3 ! 

Will wait for the released tag to update and merge.

File: Podfile.lock (hunk 1 above)
Comment (marecar3) [reply Podfile.lock]: Thanks, @etoledom!

**Pull request merged**